### DISCUSSION TITLE
Joomla 4 Code freeze

### DISCUSSION THREAD

**Discussion by wilsonge** [joomla/joomla-cms#34944] [Show and tell]
As of the release of Release Candidate 5, the Joomla 4.0 branch is going to be put into a full code freeze aside for addressing critical issues with an eye to releasing a stable release in the near future. Keep your eyes glued on social media for more information over the coming weeks.

We will be creating a 4.0.1 branch to continue merging regular bug fixes during this period.

**Comment by brianteeman** [comment]
Can you please ensure that all npm and composer packages are up to date. Especially those that are used directly such as #34888

**Reply by wilsonge** [reply]
On the roadmap to RC6 is tagging Framework stable 2.0 so yes at that point we will be updating most our direct dependencies on a composer end. npm is largely updated from a security issues perspective (the one exception is the `watch` package dragging some stuff in but that's another build tools refactor that will likely have to wait until post stable). I think it's largely going to depend on a package by package basis and what the risks are at this point. Some packages may also wait until 4.0.1 too depending on what they are how deep they are in the stack.

**Reply by brianteeman** [reply]
The reference pr is codemirror.

**Reply by wilsonge** [reply]
Yup and relatively simple things like that are obviously fine and I will merge.

**Comment by laoneo** [comment]
How do you define "critical"?
